src/couchdb/bulk-get.ts

```diff
@@ -0,0 +1,45 @@
+import nano from 'nano'
+
+export interface BulkGetRequestDoc {
+  id: string
+  rev?: string
+}
+
+export interface BulkGetResponse<T> {
```

Comment:
I don't like exporting deep type hierarchies like this. It makes too many "implementation details" part of the public contract. Solutions include:

- Only export the top-level `BulkGetResponse`, and leave the other named types private to the module.
- Flatten away the type hierarchy, so it's easier to read too:

```
export interface BulkGetResponse<T> {
  results: Array<{
    id: string
    docs: Array<{ ok: T } | { error: BulkGetError }>
  }>
}
```

I feel like, as a human, this second version is *much* easier to parse visually & understand.

Reply:
Thanks.